sigsev with forgejo and oauth2 token

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x87a0c8]

goroutine 1 [running]:
github.com/updatecli/updatecli/pkg/plugins/resources/gitea/pullrequest.(*Gitea).isPullRequestExist(0x400037e2a0)
        /home/runner/work/updatecli/updatecli/pkg/plugins/resources/gitea/pullrequest/utils.go:38 +0x1b8
github.com/updatecli/updatecli/pkg/plugins/resources/gitea/pullrequest.(*Gitea).CreateAction(0x400037e2a0, {{0x40012a7b40, 0x40}, {0x4000e657f0, 0xe}, {0x4000e657f0, 0xe}, {0x0, 0x0}, {0x40012f66e0, ...}, ...}, ...)
        /home/runner/work/updatecli/updatecli/pkg/plugins/resources/gitea/pullrequest/create.go:37 +0x15c
github.com/updatecli/updatecli/pkg/core/pipeline.(*Pipeline).RunActions(0x4001060750)
        /home/runner/work/updatecli/updatecli/pkg/core/pipeline/actions.go:150 +0x934
github.com/updatecli/updatecli/pkg/core/engine.(*Engine).runActions(0x54ff460)
        /home/runner/work/updatecli/updatecli/pkg/core/engine/actions.go:22 +0x1c4
github.com/updatecli/updatecli/pkg/core/engine.(*Engine).Run(0x54ff460)
        /home/runner/work/updatecli/updatecli/pkg/core/engine/run.go:25 +0x290
github.com/updatecli/updatecli/cmd.run({0x2fc718e?, 0x0?})
        /home/runner/work/updatecli/updatecli/cmd/root.go:101 +0xb98
github.com/updatecli/updatecli/cmd.init.func5(0x54b1d40?, {0x40009826c0?, 0x0?, 0x2?})
        /home/runner/work/updatecli/updatecli/cmd/apply.go:47 +0x234
github.com/spf13/cobra.(*Command).execute(0x54b1d40, {0x40009826a0, 0x2, 0x2})
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:989 +0x81c
github.com/spf13/cobra.(*Command).ExecuteC(0x54b11c0)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1117 +0x344
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1041
github.com/updatecli/updatecli/cmd.Execute()
        /home/runner/work/updatecli/updatecli/cmd/root.go:49 +0x68
main.main()
        /home/runner/work/updatecli/updatecli/main.go:8 +0x1c
```

### Expected Behavior

Pull request opened

### Steps To Reproduce


```
actions: gitea: {
	scmid: "forgejo"
	kind: "gitea/pullrequest"
}
```

### Environment

```markdown
- OS:

cat /etc/os-release 
NAME="Alpine Linux"
ID=alpine
VERSION_ID=3.20.3
PRETTY_NAME="Alpine Linux v3.20"
HOME_URL="https://alpinelinux.org/"
BUG_REPORT_URL="https://gitlab.alpinelinux.org/alpine/aports/-/issues"

- updatecli:
updatecli version

VERSION

Application:    0.85.0
Golang     :    1.23.2 linux/amd64
Build Time :    2024-10-07T06:48:15Z
```


### Anything else?

https://forgejo.org

While forejo is a fork, I would imagine the pull request api wouldn't have diverged.
Worst case, I would expect updatecli to return a bit of an error about what happened, instead of crashing entirely

--- a/pkg/plugins/resources/gitea/pullrequest/utils.go
+++ b/pkg/plugins/resources/gitea/pullrequest/utils.go
@@ -35,7 +35,7 @@ func (g *Gitea) isPullRequestExist() (bool, error) {
 		)
 
 		if err != nil {
-			logrus.Debugf("RC: %d\nBody:\n%s", resp.Status, resp.Body)
+			logrus.Debugf("RC: %s\n", err)
 			return false, err
 		}
 

